core/cache/controller.go

```diff
@@ -138,6 +138,20 @@ func (c *Controller) Model(uuid string) (*Model, error) {
 	return model, nil
 }
 
+// Application returns the application for the input model UUID and app name.
+// If the model or application are not found, a NotFoundError is returned.
+func (c *Controller) Application(modelUUID, appName string) (*Application, error) {
```

Comment:
Do you think it is worthwhile adding this to the Controller type, or just the Model? Personally I would have thought having it on the Model would be more correct. What are the use cases you are looking to solve with this?

Reply:
I did this after looking at the cached controller usage in the logging API server. The server context has a controller instance.

I just figure it gives you a one-liner to get an application from the controller, rather than: get the model, check for err, get the app, check for err.